### ISSUE TITLE
TDM and studio provide advance watching system, teacher system

### ISSUE THREAD

**Issue by mbonani** [Mobsya/aseba#307]
Having a studio that can watch another robot without programming it is great, but this feature have to also share the code between the studio. Probably TDM and studio should share aesl files (for instance autosave code when play button is push and transmit it to watching instance).
Probably this mechanism can be aslo used for teacher monitoring, we can imagine a special teacher version who can take fully control of robot and code to propose correction.

**Comment by mbonani** [comment]
I think the first step made in studio to connect in watching mode is great, just missing the code

**Comment by cor3ntin** [comment]
#378 Permit applications to broadcast the code to other apps.
A teacher system will require a more refined permission system

**Comment by mbonani** [comment]
Yes, I think now we can show the teacher concretely the mechanism, we have to see what is really needed as permission system. Sometimes putting to much constrain is not productive. Most teacher are the master of their classes and do not need more, other perhaps prefer technology to helps them. It is also possible that teacher want also that student can watch and exchange the code of other.  It is time to   make a survey
